components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIDefinitionFromSwagger20.java

```diff
@@ -72,12 +76,89 @@
 public class APIDefinitionFromSwagger20 implements APIDefinition {
 
     private static final Logger log = LoggerFactory.getLogger(APIDefinitionFromSwagger20.class);
+    private static Map<String, Map<String, String>> localConfigMap = new HashMap<>();
+
+    @Override
+    public String getScopeOfResourcePath(String resourceConfigsJSON, Request request,
+                                         ServiceMethodInfo serviceMethodInfo) throws APIManagementException {
+        SwaggerParser swaggerParser = new SwaggerParser();
+        Swagger swagger = swaggerParser.parse(resourceConfigsJSON);
+        String basepath = swagger.getBasePath();
+        String verb = (String) request.getProperty("HTTP_METHOD");
+//        Method resourceMethod = (Method) request.getProperty("method");todo change to this if msf4j2.3.0-m2 or higher
+        Method resourceMethod = serviceMethodInfo.getMethod();
+
+        if (resourceMethod == null || verb == null) {
+            String message = "Could not read required properties from HTTP Request. HTTP_METHOD=" + verb +
+                    " resourceTemplate=" + resourceMethod;
+            log.error(message);
+            throw new APIManagementException(message, ExceptionCodes.SWAGGER_URL_MALFORMED);
+        }
+        String apiPrefix = resourceMethod.getDeclaringClass().getAnnotation(javax.ws.rs.ApplicationPath.class).value();
+        String pathTemplate = "";
+        if (resourceMethod.getAnnotation(javax.ws.rs.Path.class) != null) {
+            pathTemplate = resourceMethod.getAnnotation(javax.ws.rs.Path.class).value();
+        }
+        String nameSpace = null;
+        if (basepath.contains("publisher")) {
+            nameSpace = APIMgtConstants.NAMESPACE_PUBLISHER_API;
+        } else if (basepath.contains("store")) {
+            nameSpace = APIMgtConstants.NAMESPACE_STORE_API;
+        } else if (basepath.contains("admin")) {
+            nameSpace = APIMgtConstants.NAMESPACE_ADMIN_API;
+        }
+
+        //if namespace is not available in local cache add it.
+        if (!localConfigMap.containsKey(nameSpace)) {
+            localConfigMap.put(nameSpace, new HashMap<>());
+        }
+
+        if (nameSpace != null && localConfigMap.get(nameSpace).isEmpty()) {
```

Comment:
it won't result in a NPE but this null check is to avoid populating Map if nameSpace is null.